flexget/plugins/clients/rtorrent.py

```diff
@@ -513,52 +496,56 @@ def on_task_download(self, task, config):
-                    entry.fail("failed to render properties %s" % str(e))
+        try:
+            for entry in task.accepted:
+                if config['action'] == 'add':
+                    if task.options.test:
+                        log.info('Would add %s to rTorrent', entry['url'])
+                        continue
+                    try:
+                        options = self._build_options(config, entry)
+                    except RenderError as e:
+                        entry.fail("failed to render properties %s" % str(e))
+                        continue
+
+                    # fast_resume is not really an rtorrent option so it's not in _build_options
+                    fast_resume = entry.get('fast_resume', config['fast_resume'])
+                    self.add_entry(client, entry, options, start=config['start'], mkdir=config['mkdir'],
+                                   fast_resume=fast_resume)
+
+                info_hash = entry.get('torrent_info_hash')
+
+                if not info_hash:
+                    entry.fail('Failed to %s as no info_hash found' % config['action'])
                     continue
 
-                # fast_resume is not really an rtorrent option so it's not in _build_options
-                fast_resume = entry.get('fast_resume', config['fast_resume'])
-                self.add_entry(client, entry, options, start=config['start'], mkdir=config['mkdir'],
-                               fast_resume=fast_resume)
+                if config['action'] == 'delete':
+                    if task.options.test:
+                        log.info('Would delete %s (%s) from rTorrent', entry['title'], entry['torrent_info_hash'])
+                        continue
+                    self.delete_entry(client, entry)
 
-            info_hash = entry.get('torrent_info_hash')
+                if config['action'] == 'update':
+                    if task.options.test:
+                        log.info('Would update %s (%s) in rTorrent', entry['title'], entry['torrent_info_hash'])
+                        continue
+                    self.update_entry(client, entry, config)
 
-            if not info_hash:
-                entry.fail('Failed to %s as no info_hash found' % config['action'])
-                continue
-
-            if config['action'] == 'delete':
-                if task.options.test:
-                    log.info('Would delete %s (%s) from rTorrent', entry['title'], entry['torrent_info_hash'])
-                    continue
-                self.delete_entry(client, entry)
-
-            if config['action'] == 'update':
-                if task.options.test:
-                    log.info('Would update %s (%s) in rTorrent', entry['title'], entry['torrent_info_hash'])
-                    continue
-                self.update_entry(client, entry, config)
+        except IOError as e:
+            raise plugin.PluginError("Couldn't connect to rTorrent: %s" % str(e))
```

Comment:
I don't think `str` is necessary.

Reply:
I didn't change this, it's a move of the code. I think we don't do str the message comes out differently

Reply:
maybe you are right, i just tested it. Are we sure this is not a py3 thing?